### PR TITLE
Validate domains front & back.

### PR DESCRIPTION
**Pull request by mateusz**
domval

### CODE REVIEW

File: js/ServerNames.jsx [hunk 1]
```diff
@@ -91,6 +91,26 @@ var EditForm = React.createClass({
 		});
 	},
 
+	validateDomain: function(value) {
+			var message = '';
```
Comment (halkyon) [inline js/ServerNames.jsx]: Double tabbing!

File: js/ServerNames.jsx [hunk 2]
```diff
@@ -138,8 +156,9 @@ var EditForm = React.createClass({
 							<form.Input
 								name="PrimaryDomain"
 								value={this.state.PrimaryDomain}
+								validations='isFQDN|{"require_tld":false}'
 								required
-								validationError="Please enter primary domain name" />
+								validationError='Invalid domain. Only alhpanumerics, dashes and dots are permitted, and the domain must begin and end with alphanumeric.' />
```
Comment (halkyon) [inline js/ServerNames.jsx]: alhpanumerics is mispelled

File: code/EnvironmentConfig/DomainsDispatcher.php [hunk 3]
```diff
@@ -87,29 +86,51 @@ public function getModel($name) {
 	}
 
 	/**
+	 * Check if the domain is a valid hostname.
+	 *
+	 * @param string $domain
+	 * @return bool
+	 */
+	public function validateDomain($domain) {
```
Comment (halkyon) [inline code/EnvironmentConfig/DomainsDispatcher.php]: Unit test? :)

File: js/ServerNames.jsx (hunk 1 above)
Comment (mateusz) [inline js/ServerNames.jsx]: fixed

File: js/ServerNames.jsx (hunk 2 above)
Comment (mateusz) [inline js/ServerNames.jsx]: fixed

File: code/EnvironmentConfig/DomainsDispatcher.php (hunk 3 above)
Comment (mateusz) [inline code/EnvironmentConfig/DomainsDispatcher.php]: fixed

**Pull request merged**